journal/202206.md

```diff
@@ -0,0 +1,491 @@
+# {DRAFT} Decred Journal – June 2022
+
+{ _Only remove DRAFT from the title when all other todos are solved, at the very end. Search for `{` to see what's left._ }
+
+{ _Hint: Hints just add context. If it doesn't start with "Hint: ", it is a task. This was a hint._ }
+
+{ _Hint: Read the Guidelines on how to make DJ great: https://github.com/xaur/decred-news/blob/docs/guidelines.md ._}
+
+![{alt text}](../img/{file.ext} "{hover tooltip text}")
+
+_Image: { "Title by @author. Description..." OR just "Description..." }_
+
+{ _Month's highlights, no links here, 5-7 bullets max._
+
+- Ethereum smart contract used in DCRDEX passed its first audit with great results.
+- 
+- 
+- 
+- 
+
+}
+
+Contents: { _Sync ToC at the very end, when all text is finished_ }
+
+- [Development](#development)
+- [People](#people)
+- [Governance](#governance)
+- [Network](#network)
+- [Ecosystem](#ecosystem)
+- [Outreach](#outreach)
+- [Events](#events)
+- [Media](#media)
+- [Discussions](#discussions)
+- [Markets](#markets)
+- [Relevant External](#relevant-external)
+
+{ _Add individual sections for high-impact announcements before Development, such as consensus votes, software releases, milestones, major bugs, etc._ }
+
+
+## Development
+
+The work reported below has the "merged to master" status unless noted otherwise. It means that the work is completed, reviewed, and integrated into the source code that advanced users can [build and run](https://medium.com/@artikozel/the-decred-node-back-to-the-source-part-one-27d4576e7e1c), but is not yet available in release binaries for regular users.
+
+
+### dcrd
+
+_[dcrd](https://github.com/decred/dcrd) is a full node implementation that powers Decred's peer-to-peer network around the world._
+
+- [Optimized](https://github.com/decred/dcrd/pull/2957) cumulative work calculation using the new zero-allocation [256-bit integers](https://github.com/decred/dcrd/pull/2787) package. The result is ~100 MiB less heap usage on average, and ~5% faster initial blockchain sync time.
+- Support [clean shutdown](https://github.com/decred/dcrd/pull/2958) on more Unix variants and on Windows (triggered by events like user logging off, closing terminal window, or system shutdown).
+- [Inverted order](https://github.com/decred/dcrd/pull/2956) in which transactions are added to the mempool during a reorg to correct the calculation of transaction chain statistics under special circumstances. { _Did it happen in the wild? What was the impact?_ }
+- [Removed](https://github.com/decred/dcrd/pull/2961) the spend journal pruning code that became unneeded with the removal of the address index.
+- Explicitly [reject](https://github.com/decred/dcrd/pull/2963) standalone `treasurybase` transactions. { _Looks like a minor change to merge into "other", but to understand this I wonder what would be a simple description of "proper" (i.e. "not standalone") `treasurybase` transactions?_ }
```

Comment:
Other is fine.  The behavior didn't actually change since treasurybases are already implicitly rejected as the PR notes.  The PR in question just does it more explicitly as a form of defensive coding in case future changes violate those implicit assumptions.  Explicit > implicit.

As far as "proper" treasurybases, as of the activation of [DCP0006](https://github.com/decred/dcps/blob/master/dcp-0006/dcp-0006.mediawiki), every block is required to have a treasurybase transaction as the first transaction in the stake tree that pays the required subsidy to the Treasury.  It mirrors the coinbase in the regular the transaction tree that pays the required subsidy to the PoW miners.

Neither coinbases nor treasurybases make sense as standalone (aka unconfirmed, not part of a block) transactions because they are only valid when part of the block generation process.

Reply:
Woah, every time I learn something new. Added for education. Thanks.